docs/topics/proc_configuring-jenkins-to-deploy-an-application-with-fmp.adoc

```diff
@@ -0,0 +1,40 @@
+= Configuring Jenkins to deploy your application with the Fabric8 Maven Plugin
+
+Similar to using maven and the Fabric8 Maven Plugin from your local host to deploy an application, you can configure Jenkins to use maven and the Fabric8 Maven Plugin to deploy an application.
+
+[discrete]
+== Prerequisites
+* Access to an OpenShift cluster.
+* Jenkins installed and configured to access the same OpenShift cluster.
+* JDK and Maven installed and configured on your Jenkins server.
+* An application available in GitHub and configured to use maven and the Fabric8 Maven Plugin.
```

Comment:
The prerequisites are pretty extensive. I mean, especially the 2nd one -- "Jenkins installed and configured to access the same OpenShift cluster". It's by no means clear if that expects Jenkins installed inside OpenShift or outside, but that makes a lot of difference later in the process.

Reply:
I'm for clarifying point 2, but the rest seems good to me. These are the general requirements, and either we expect a specific procedure to be present earlier in the assembly that tells you how to do that specific thing, or a general knowledge how to do it if this module is presented by itself (not a big use case for the latter, we'll publish everything in assemblies primarily).

Reply:
Yea, right, my objection is mainly against point 2. Point 3 could use a bit of explanation as well -- you do need to do it in a traditional Jenkins setup, but if you deploy Jenkins to OpenShift, or at least let it create agents on OpenShift, you don't really have to care about installing JDK and Maven in Jenkins, you only have to make sure the Jenkins agent image has them.

Reply:
@Ladicek re: JDK and Maven on jenkins in OpenShift, I used minishift and the Jenkins instant app to test this and I still had to install a JDK and configure JAVA_HOME for the FMP builds to work.

Reply:
Not sure what's a "Jenkins instant app", but I'm pretty sure the way OpenShift sets up Jenkins for the pipelines (wihch I replicate in Rourka) doesn't need that. All builds happen in agent pods, which are spun up by the Kubernetes Jenkins plugin, using by default OpenShift-provided images, which already have Java and Maven (or Node.js) installed.

But then -- the fact that we're having this discussion shows that it's more complex than just one sentence :-)

I don't necessarily think RHOAR docs must explain all this (or do they?). But it's complex.